Build issue with Spack

I'm trying to build tioga for nalu-wind.  However, I'm getting the following error when I try to build with gcc 7:

```
     129    [ 72%] Building Fortran object driver/CMakeFiles/tioga.exe.dir/testTioga.F90.o
     130    cd /tmp/psakiev/spack-stage/spack-stage-tioga-develop-2654ulebty7znslynmhhia2wmyxz3p45/spack-build-2654ule/driver && /pro
            jects/wind/spack-manager/spack/lib/spack/env/gcc/gfortran -D-pthread -DTIOGA_HAS_NODEGID -I/sierra/sntools/SDK/mpi/openmp
            i/4.0.5-gcc-7.2.0-RHEL7/include -I/projects/sierra/linux_rh7/SDK/mpi/openmpi/4.0.5-gcc-7.2.0-RHEL7/include -I/sierra/snto
            ols/SDK/mpi/openmpi/4.0.5-gcc-7.2.0-RHEL7/lib -I/tmp/psakiev/spack-stage/spack-stage-tioga-develop-2654ulebty7znslynmhhia
            2wmyxz3p45/spack-src/src -fbounds-check -fbacktrace -fdefault-real-8 -ffree-line-length-none -O3 -DNDEBUG -O3 -fPIE -cpp 
            -c /tmp/psakiev/spack-stage/spack-stage-tioga-develop-2654ulebty7znslynmhhia2wmyxz3p45/spack-src/driver/testTioga.F90 -o 
            CMakeFiles/tioga.exe.dir/testTioga.F90.o
     131    <command-line>:1:0:
     132    
  >> 133    Error: macro names must be identifiers
  >> 134    make[2]: *** [driver/CMakeFiles/tioga.exe.dir/testTioga.F90.o] Error 1
     135    make[2]: Leaving directory `/tmp/psakiev/spack-stage/spack-stage-tioga-develop-2654ulebty7znslynmhhia2wmyxz3p45/spack-bui
            ld-2654ule'
  >> 136    make[1]: *** [driver/CMakeFiles/tioga.exe.dir/all] Error 2
     137    make[1]: Leaving directory `/tmp/psakiev/spack-stage/spack-stage-tioga-develop-2654ulebty7znslynmhhia2wmyxz3p45/spack-bui
            ld-2654ule'
  >> 138    make: *** [all] Error 2
  ```
pretty sure the issue is the flag `-D-pthread` being fed to the fortran compiler.  I can't seem to track down where this flag is being created. So far I've only seen this on rhel7 and gcc7. I don't see this issue when I build on mac-os with gcc 8., but the `-D-pthread` flag is missing in that build. Those are the only two complier combinations I've tested this on so far.

Thanks @svdavidson. I'm not sure what the differences are between the two branches. I'll have to look at them closer.  I'm just starting to look at tioga for the first time. I ran into another build error with `ifort` this morning.  If I revert the last commit on the `exawind` branch it builds fine. I haven't had a chance to look to0 closely at what is causing this yet.  

```
3 errors found in build log:
     133    [ 72%] Building Fortran object driver/CMakeFiles/poisson_mms.exe.dir/poisson_mms.f90.o
     134    cd /projects/wind/spack-manager/spack/var/spack/stage/psakiev/spack-stage-tioga-develop-ihz7xycwe5psq2mao5z2pjrfu2cjnhss/spack-build-ihz7xyc/driver && /projects/wind/spack-manager/sp
            ack/lib/spack/env/intel/ifort -DTIOGA_HAS_NODEGID -I/opt/openmpi/4.0/intel/include -I/opt/openmpi/4.0/intel/lib -I/projects/wind/spack-manager/spack/var/spack/stage/psakiev/spack-sta
            ge-tioga-develop-ihz7xycwe5psq2mao5z2pjrfu2cjnhss/spack-src/src -r8 -double_size 128 -O3 -fPIE -c /projects/wind/spack-manager/spack/var/spack/stage/psakiev/spack-stage-tioga-develop
            -ihz7xycwe5psq2mao5z2pjrfu2cjnhss/spack-src/driver/poisson_mms.f90 -o CMakeFiles/poisson_mms.exe.dir/poisson_mms.f90.o
     135    [ 74%] Linking CXX executable poisson_mms.exe
     136    cd /projects/wind/spack-manager/spack/var/spack/stage/psakiev/spack-stage-tioga-develop-ihz7xycwe5psq2mao5z2pjrfu2cjnhss/spack-build-ihz7xyc/driver && /projects/wind/spack-manager/sp
            ack/opt/spack/linux-rhel7-sandybridge/intel-19.0.3.199/cmake-3.20.3-uno43e5pjqawx232xmxscgo5mpjervrw/bin/cmake -E cmake_link_script CMakeFiles/poisson_mms.exe.dir/link.txt --verbose=
            1
     137    /projects/wind/spack-manager/spack/lib/spack/env/intel/icpc -O3 -DNDEBUG CMakeFiles/poisson_mms.exe.dir/poisson_mms.f90.o -o poisson_mms.exe  -Wl,-rpath,/opt/openmpi/4.0/intel/lib:::
            ::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
            ::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::: libtiogadriver.a ../src/libtioga.a /opt/openmpi/4.0/intel/lib/libmpi_usempif08.so /opt/ope
            nmpi/4.0/intel/lib/libmpi_usempi_ignore_tkr.so /opt/openmpi/4.0/intel/lib/libmpi_mpifh.so /opt/openmpi/4.0/intel/lib/libmpi.so -ldl -lifport -lifcoremt -lpthread
     138    /lib/../lib64/crt1.o: In function `_start':
  >> 139    (.text+0x20): undefined reference to `main'
  >> 140    make[2]: *** [driver/CMakeFiles/poisson_mms.exe.dir/build.make:105: driver/poisson_mms.exe] Error 1
     141    make[2]: Leaving directory '/projects/wind/spack-manager/spack/var/spack/stage/psakiev/spack-stage-tioga-develop-ihz7xycwe5psq2mao5z2pjrfu2cjnhss/spack-build-ihz7xyc'
  >> 142    make[1]: *** [CMakeFiles/Makefile2:172: driver/CMakeFiles/poisson_mms.exe.dir/all] Error 2
     143    make[1]: Leaving directory '/projects/wind/spack-manager/spack/var/spack/stage/psakiev/spack-stage-tioga-develop-ihz7xycwe5psq2mao5z2pjrfu2cjnhss/spack-build-ihz7xyc'
     144    make: *** [Makefile:139: all] Error 2
```

--- a/driver/CMakeLists.txt
+++ b/driver/CMakeLists.txt
@@ -12,8 +12,16 @@ if (TIOGA_ENABLE_ARBORX)
 endif()
 
 add_library(tiogadriver ${TIOGA_EXE_SOURCES})
-target_compile_options(tiogadriver PUBLIC ${MPI_Fortran_COMPILE_FLAGS})
+
+set(libopts ${MPI_fortran_COMPILE_FLAGS})
+
+if( CMAKE_Fortran_COMPILER_ID STREQUAL "Intel" )
+    list(APPEND libopts "-nofor-main")
+endif()
+
+target_compile_options(tiogadriver PUBLIC ${libopts})
 target_include_directories(tiogadriver PUBLIC ${MPI_Fortran_INCLUDE_PATH})
+
 target_link_libraries(tiogadriver INTERFACE
   tioga ${MPI_Fortran_LIBRARIES} ${CMAKE_DL_LIBS})
 if (TIOGA_ENABLE_ARBORX)

--- a/driver/CMakeLists.txt
+++ b/driver/CMakeLists.txt
@@ -12,7 +12,7 @@ if (TIOGA_ENABLE_ARBORX)
 endif()
 
 add_library(tiogadriver ${TIOGA_EXE_SOURCES})
-target_compile_definitions(tiogadriver PUBLIC ${MPI_Fortran_COMPILE_FLAGS})
+target_compile_options(tiogadriver PUBLIC ${MPI_Fortran_COMPILE_FLAGS})
 target_include_directories(tiogadriver PUBLIC ${MPI_Fortran_INCLUDE_PATH})
 target_link_libraries(tiogadriver INTERFACE
   tioga ${MPI_Fortran_LIBRARIES} ${CMAKE_DL_LIBS})

--- a/gridGen/CMakeLists.txt
+++ b/gridGen/CMakeLists.txt
@@ -8,7 +8,7 @@ set(GRIDGEN_SOURCES
 
 add_executable(buildGrid ${GRIDGEN_SOURCES})
 
-target_compile_definitions(buildGrid PUBLIC ${MPI_Fortran_COMPILE_FLAGS})
+target_compile_options(buildGrid PUBLIC ${MPI_Fortran_COMPILE_FLAGS})
 target_include_directories(buildGrid PUBLIC ${MPI_Fortran_INCLUDE_PATH})
 target_link_libraries(buildGrid
   tioga ${MPI_Fortran_LIBRARIES} ${CMAKE_DL_LIBS})

